### ISSUE TITLE
archinstall tries installing on import.

### ISSUE THREAD

**Issue by git-girl** [archlinux/archinstall#2728]
### Which ISO version are you using?

(git latest but also 2.8.6-1 but since longer afaik)

### The installation log

```json
this is not about that.
```

#### Description of the issue

importing archinstall 

```
git clone https://github.com/archlinux/archinstall.git
cd archinstall
python
import archinstall
```

archinstall will try to do an installation. 
=>
```
/dev/mapper/control: open failed: Permission denied
Failure to communicate with kernel device-mapper driver.
Incompatible libdevmapper 1.02.201 (2024-10-02) and kernel driver (unknown version).
Not enough permission to place log file at /var/log/archinstall/install.log, creating it in /tmp/archinstall/install.log instead
Archinstall requires root privileges to run. See --help for more.
```
(also just on archinstall --version)

Imo this is especially annoying, bc there is tooling such as pydoc that when just searching with `-k` will evaluate everything on the system. So when i go to look at docs archinstall tries to do an archinstall. If I were a bit more stupid then i already am and maybe run `pydoc -k something` as euid 0 then i do archinstall? (it just crashes ftr)

**Comment by svartkanin** [comment]
It isn't doing an installation, it's trying to read your available devices which requires root privileges. What hppens if you run the above in a root shell?

**Comment by Torxed** [comment]
I agree that simply importing the library shouldn't trigger things. 
It's fine for what we use it for so I've let it slide for a few years now. 

We should change this behavior, but I'll be upfront and say that this is a low priority fix personally.

**Comment by svartkanin** [comment]
We can definitely lazy load it, but generally archinstall will not work without root, so even if lazy loaded it may break shortly after anyways

**Comment by Torxed** [comment]
In the old days, dinosaur times, the tool could also be used as Python lib to do package search and stuff. If you wanted to avoid using `subprocess` yourself.
But it's a very small use case that doesn't require root that's for sure.
